src/AzureAuth.Test/CommandMainTest.cs

```diff
@@ -503,6 +509,167 @@ public void TestCacheFileOptionWithWindowsAbsolutePath()
             subject.CacheFilePath.Should().Be(path);
         }
 
+        /// <summary>
+        /// Test to generate event data from a null authflow result.
+        /// </summary>
+        [Test]
+        public void TestGenerateEvent_FromNullAuthResult()
+        {
+            AuthFlowResult authFlowResult = null;
+            var subject = this.serviceProvider.GetService<CommandMain>();
+
+            // Act
+            var eventData = subject.GenerateEventData(authFlowResult);
+
+            // Assert
+            eventData.Should().BeNull();
+        }
+
+        /// <summary>
+        /// Test to generate event data from an authflow result with null token result and null errors.
+        /// </summary>
+        [Test]
+        public void TestGenerateEvent_From_AuthFlowResult_With_Null_TokenResult_Null_Errors()
+        {
+            AuthFlowResult authFlowResult = new AuthFlowResult(null, null, "AuthFlowName");
+            var subject = this.serviceProvider.GetService<CommandMain>();
+
+            // Act
+            var eventData = subject.GenerateEventData(authFlowResult);
+
+            // Assert
+            eventData.Properties.Should().NotContainKey("msal_correlation_ids");
+            eventData.Properties.Should().NotContainKey("error_messages");
+            eventData.Measures.Should().NotContainKey("token_validity_hours");
+            eventData.Properties.Should().NotContainKey("is_silent");
+
+            eventData.Properties.TryGetValue("auth_mode", out string authMode);

```

Comment:
Don't quote me on this, but from looking at the [Fluent Assertions docs](https://fluentassertions.com/dictionaries/) I _think_ you might be able to write this as
```csharp
eventData.Properties.Should().Contain("auth_mode", "AuthFlowName");
```

Reply:
done.